src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/queues/common.kt

```diff
@@ -202,7 +205,13 @@ fun refundTransaction(
         } else {
           Mono.just(tx)
         }
-        .flatMap { refundRetryService.enqueueRetryEvent(it, retryCount) }
+        .flatMap {
+          when(exception){
+            // Enqueue retry event only if refund is allowed
+            !is RefundNotAllowedException -> refundRetryService.enqueueRetryEvent(it, retryCount)
```

Comment:
add junit test to cover the case when retry event is enqueued and when is not based on the PGS received http response code

Reply:
You're right, done with [4db90e8](https://github.com/pagopa/pagopa-ecommerce-event-dispatcher-service/pull/109/commits/4db90e81add31093172d938690cb7875eebfa5e6) 👍